### PR TITLE
Added cluster-group-upgrade package

### PR DESCRIPTION
**Pull request by shaior**

### CODE REVIEW

File: pkg/cgu/cgu.go [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+					"Failed to update the cgu object %s, "+
+						"due to error in delete function", builder.Definition.Name)
+
+				return nil, err
+			}
+
+			return builder.Create()
+		}
+	}
+
```
Comment (trewest) [inline pkg/cgu/cgu.go:175]:
```suggestion
	if err == nil {
		builder.Object = builder.Definition
	}
```

File: pkg/cgu/list.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/cluster-group-upgrades-operator/api/v1alpha1"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ListCgusInAllNamespaces returns a cluster-wide cgu inventory.
+func ListCgusInAllNamespaces(apiClient *clients.Settings, options metaV1.ListOptions) ([]*CguBuilder, error) {
```
Comment (trewest) [inline pkg/cgu/list.go:14]:
```suggestion
func ListInAllNamespaces(apiClient *clients.Settings, options metaV1.ListOptions) ([]*CguBuilder, error) {
```

Comment (trewest) [inline pkg/cgu/list.go:14]: Please use the options parameter that is being provided

File: pkg/cgu/list.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	glog.V(100).Info("Listing all CGUs in all namespaces")
+
+	policyList := &v1alpha1.ClusterGroupUpgradeList{}
```
Comment (trewest) [inline pkg/cgu/list.go:17]:
```suggestion
	cguList := &v1alpha1.ClusterGroupUpgradeList{}
```

**Review by trewest — CHANGES_REQUESTED** [review]
Few comments

**Review by trewest — CHANGES_REQUESTED** [review]
Just one change and we can merge

**Comment by trewest** [comment]
@shaior can you resolve the merge conflicts? Other than that, looks good. Waiting for one more approval

**Review by cdvultur — CHANGES_REQUESTED** [review]
Small filename change, for consistency
Otherwise it looks great !

**Review by cdvultur — APPROVED** [review]
/lgtm

**Review by trewest — APPROVED** [review]
/lgtm

**Pull request merged**